rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java

```diff
@@ -398,17 +400,33 @@ public String eth_blockNumber() {
     }
 
     @Override
-    public String eth_call(CallArguments args, Map<String, String> inputs) {
-        return invokeByBlockRef(inputs, blockNumber -> this.eth_call(args, blockNumber));
+    public String eth_call(CallArgumentsParam args, Map<String, String> inputs) {
+        return invokeByBlockRef(inputs, blockNumber -> this.eth_call(args, new BlockIdentifierParam(blockNumber)));
     }
 
     @Override
-    public String eth_getCode(String address, Map<String, String> inputs) {
-        return invokeByBlockRef(inputs, blockNumber -> this.eth_getCode(address, blockNumber));
+    public String eth_getCode(HexAddressParam address, BlockRefParam blockRefParam) {
+        if (blockRefParam.getIdentifier() != null) {
+            return this.getCode(address, blockRefParam.getIdentifier());
+        } else {
+            return this.eth_getCode(address, blockRefParam.getInputs());
+        }
+    }
+
+    private String eth_getCode(HexAddressParam address, Map<String, String> inputs) {
+        return invokeByBlockRef(inputs, blockNumber -> this.getCode(address, blockNumber));
     }
 
     @Override
-    public String eth_getBalance(String address, String block) {
+    public String eth_getBalance(HexAddressParam address, BlockRefParam blockRefParam) {
+        if(blockRefParam.getIdentifier() != null) {
+            return this.eth_getBalance(address, blockRefParam.getIdentifier());
+        } else {
+            return this.eth_getBalance(address, blockRefParam.getInputs());
```

Comment:
I don't think is necessary to specically check for null at the moment since the system [throws an error](https://github.com/rsksmart/rskj/blob/json_rpc_param_validation/rskj-core/src/main/java/org/ethereum/rpc/Web3Impl.java#L529) if the map does not have any of the blockref keys.